### DISCUSSION TITLE
Is there way to get MIDI input from an external USB keyboard?

### DISCUSSION THREAD

**Discussion by skybrian** [falkTX/Carla#1673] [Q&A] [answered]
I tried running Carla 2.5 on a fairly old Mac running 10.13.6 (High Sierra). The OS cannot be upgraded so newer software sometimes doesn't run, but Carla works. Except that I don't really understand it.

In particular, I don't see any way to select an external MIDI keyboard in Carla. Is this possible? Where would I configure it? Or is it supposed to just work?

Specifics: I was able to load an SFZ file and get Carla to play notes. To do this, I switched to the PatchBay tab and connected the MIDI Input component to the SFZ component, and the SFZ component to Audio Output.

If I put focus on the the SFZ component, I can play notes using the virtual keyboard below the patch bay. However, playing notes on an actual MIDI keyboard connected to the USB port has no effect. (The same keyboard works in VCV Rack after configuring it.) And I don't see a way to configure the MIDI Input component.

**Comment by falkTX** [comment, marked as answer]
Use canvas -> show external. the hw devices will be there

**Reply by falkTX** [reply]
I mean the menu bar actions

**Reply by Ricccc2k12** [reply]
doesnt show 
<img width="422" height="186" alt="image" src="https://github.com/user-attachments/assets/b1782aa7-db7e-4c32-9d03-14ab320476c2" />

**Comment by joshgura** [comment]
sorry to necrobump this, but... it would be nice if both external and internal nodes would be displayed on the patchbay at the same time. i had carla working with my Maudio keyboard before, but put carla away for a few weeks, and then came back and could not figure out how to get the keyboard to work again until finding this post on github. and maybe if "patchbay" was called "canvas" or vice versa if they are the same thing, then it would be easier to find the solution as a user. because right now there's 2 names for the same thing, and if they're different then ya coulda fooled me.
